internal/api/mfa.go

```diff
@@ -156,6 +215,65 @@ func (a *API) enrollPhoneFactor(w http.ResponseWriter, r *http.Request, params *
 	})
 }
 
+func (a *API) enrollWebAuthnFactor(w http.ResponseWriter, r *http.Request, params *EnrollFactorParams) error {
+	ctx := r.Context()
+	config := a.config
+	user := getUser(ctx)
+	session := getSession(ctx)
+	db := a.db.WithContext(ctx)
+	factors := user.Factors
+
+	factorCount := len(factors)
+	numVerifiedFactors := 0
+	if err := models.DeleteExpiredFactors(db, config.MFA.FactorExpiryDuration); err != nil {
+		return err
+	}
+	for _, factor := range user.Factors {
+		switch {
+		case factor.FriendlyName == params.FriendlyName:
```

Comment:
Validation checks can be condensed into a single function. Work for that is done here: https://github.com/supabase/auth/pull/1780

Reply:
Mind rebasing this on top of it, so the change is a bit clearer.